Snek ASM Incorporation

Snakey11 opened the pull request:
Includes Anima Triangle (Resolves Tri-Adept conflitc), Modular EXP (Resolves Paragon conflict), Portrait Selection (Resolves RandomMugs3 conflict), Stairs (Integration into MSG mov and unit menu), and Raid (Unit menu). Also adds sound effects for Poison Strike/Grisly Wound because oops I forgot to do that.

StanHash commented on Engine Hacks/Raid/Raid.event:
```diff
@@ -0,0 +1,16 @@
+	
+	#else
+
+		WARNING "You are trying to install Raid multiple times. If you are using the Skill System, go to Engine Hacks/Config.event and install it there only."
```
Installing a hack "multiple times" shouldn't be an error or warning.

StanHash commented on Engine Hacks/MSS/asm/mss_page1_skills.s:
```diff
@@ -114,7 +114,7 @@ draw_growth_at 17, 13
 ldr		r0,[r0]			@hp growth getter (not displaying because there's no room atm)
 draw_growth_at 17, 15
-draw_textID_at 13, 15, textID=0x4E9, growth_func=1, width=2 @hp name
+draw_textID_at 13, 15, textID=0x4E9, growth_func=1 @hp name
```
no

StanHash commented on Engine Hacks/Config.event:
```diff
@@ -23,6 +23,29 @@
+
+// If uncommented, Raid will be implemented.
+// This patch adds an entry to the Unit Menu, so installing here is necessary.
+// #define ENABLE_RAID
```
I'm honestly not convinced about this whole "using `Config.event` for hack installation switches" thing. I think it'd better if one could just (un)comment an `#include` in the master hack installer, and then use defines from within that hack installer for any further ifdef guards.

`ENABLE_SLAYER_AND_EFFECTIVENESS_REWORK` and `USE_SAVE_EXPANSION` were there because they're switches between different implementations of the same thing (effectiveness rework being about nullify), not just "include or not" options.

Snakey11 commented:
I've added 6 new commits that do the following:

- Change .asm files to .s files.
- Use of #ifndef guards over ENABLE_... definitions for modular hack installation. (i.e. using `STAIRS_ASM` instead of `ENABLE_STAIRS` when determining whether to write that to the unit menu)
- Getting rid of the ENABLE_... stuff altogether. (Just using regular #ifndef guards)
- Reorganized calculation loops into their own folders/installers.
- Added `PROTECT` commands to calculation loops to detect conflicts (in conjunction to doing so with my own independent hacks in my Dropbox).
- Reverted that weird MSS change and an accidental textbuildfile change.

Thanks for your input!
